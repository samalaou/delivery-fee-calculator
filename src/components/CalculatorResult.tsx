import { Grid, Typography } from "@mui/material";

interface CalculatorResultProps {
  deliveryFee: number | null;
}

function CalculatorResult({ deliveryFee }: CalculatorResultProps): JSX.Element {
  return (
    <Grid container item xs={12} style={{ margin: "20px 10px" }}>
      <Grid item xs={6}>
        <Typography variant="h6" sx={{ fontWeight: "bold", textAlign: "left" }}>
          Delivery Fee:
        </Typography>
      </Grid>
      {deliveryFee !== null && (
        <Grid item xs={5}>
          <Typography
            variant="h6"
            sx={{ fontWeight: "bold", textAlign: "right" }}
            data-test-id="fee"
          >
            {deliveryFee} euros
          </Typography>
        </Grid>
      )}
    </Grid>
  );
}

export default CalculatorResult;
