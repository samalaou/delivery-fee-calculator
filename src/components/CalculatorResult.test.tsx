import { render } from "@testing-library/react";
import CalculatorResult from "./CalculatorResult";

describe("CalculatorResult component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<CalculatorResult deliveryFee={10} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders delivery fee when deliveryFee is not null", () => {
    const { getByText } = render(<CalculatorResult deliveryFee={10} />);
    const deliveryFeeElement = getByText("10 euros");
    expect(deliveryFeeElement).toBeInTheDocument();
  });

  it("does not render delivery fee when deliveryFee is null", () => {
    const { queryByTestId } = render(<CalculatorResult deliveryFee={null} />);
    const deliveryFeeElement = queryByTestId("fee");
    expect(deliveryFeeElement).toBeNull();
  });
});
