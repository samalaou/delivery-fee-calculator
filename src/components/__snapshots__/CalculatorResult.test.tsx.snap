// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorResult component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
    style="margin: 20px 10px;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1voxdgm-MuiTypography-root"
      >
        Delivery Fee:
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-152558n-MuiTypography-root"
        data-test-id="fee"
      >
        10 euros
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
